src/test/regress/sql/background_rebalance_parallel.sql

```diff
@@ -128,13 +137,119 @@ SELECT D.task_id,
+INSERT INTO table1_colg3 SELECT i FROM generate_series(0, 100)i;
+
+CREATE TABLE  table2_colg3 (b int primary key);
+SELECT create_distributed_table('table2_colg3', 'b', colocate_with => 'table1_colg3');
+INSERT INTO table2_colg3 SELECT i FROM generate_series(0, 100)i;
+
+CREATE TABLE table1_colg4 (a int PRIMARY KEY);
+SELECT create_distributed_table('table1_colg4', 'a', shard_count => 3, colocate_with => 'none');
+INSERT INTO table1_colg4 SELECT i FROM generate_series(0, 100)i;
+
+CREATE TABLE table2_colg4 (b int PRIMARY KEY);
+SELECT create_distributed_table('table2_colg4', 'b', colocate_with => 'table1_colg4');
+INSERT INTO table2_colg4 SELECT i FROM generate_series(0, 100)i;
+
+-- Add nodes so that we can rebalance
+SELECT citus_add_node('localhost', :worker_2_port);
+SELECT citus_add_node('localhost', :worker_3_port);
+
+SELECT citus_rebalance_start AS job_id from citus_rebalance_start() \gset
+
+-- see dependent tasks to understand which tasks remain runnable because of
+-- citus.max_parallel_tasks_per_node
+-- and which tasks are actually blocked from colocation group dependencies
+SELECT D.task_id,
+       (SELECT T.command FROM pg_dist_background_task T WHERE T.task_id = D.task_id),
+       D.depends_on,
+       (SELECT T.command FROM pg_dist_background_task T WHERE T.task_id = D.depends_on)
+FROM pg_dist_background_task_depend D  WHERE job_id in (:job_id) ORDER BY D.task_id, D.depends_on ASC;
+
+-- default citus.max_parallel_tasks_per_node is 1
+SHOW citus.max_parallel_tasks_per_node;
+
+-- show that first exactly one task per node is running
+-- among the tasks that are not blocked
+SELECT citus_task_wait(1013, desired_status => 'running');
+SELECT job_id, task_id, status, nodes_involved
+FROM pg_dist_background_task WHERE job_id in (:job_id) ORDER BY task_id;
+
+-- increase citus.max_parallel_tasks_per_node
+ALTER SYSTEM SET citus.max_parallel_tasks_per_node = 2;
+SELECT pg_reload_conf();
+SELECT citus_task_wait(1015, desired_status => 'running');
+SELECT citus_task_wait(1013, desired_status => 'done');
+
+-- show that at most 2 tasks per node are running
+-- among the tasks that are not blocked
+SELECT job_id, task_id, status, nodes_involved
+FROM pg_dist_background_task WHERE job_id in (:job_id) ORDER BY task_id;
+
+-- decrease to default (1)
+ALTER SYSTEM RESET citus.max_parallel_tasks_per_node;
+SELECT pg_reload_conf();
+SELECT citus_task_wait(1015, desired_status => 'done');
+SELECT citus_task_wait(1014, desired_status => 'done');
+SELECT citus_task_wait(1016, desired_status => 'running');
+
+-- show that exactly one task per node is running
+-- among the tasks that are not blocked
+SELECT job_id, task_id, status, nodes_involved
+FROM pg_dist_background_task WHERE job_id in (:job_id) ORDER BY task_id;
+
+SELECT citus_rebalance_stop();
+-- waiting on this rebalance is racy, as it sometimes sees no rebalance is ongoing while other times it actually sees it ongoing
+-- we simply sleep a bit here
+SELECT pg_sleep(1);
```

Comment:
Looking at the code of citus_rebalance_wait this shouldn't matter as long as you set client_min_messages to ERROR.